### ISSUE TITLE
prefer-const to allow mutations of objects

### ISSUE THREAD

**Issue by antstanley** [eslint/eslint#11997]
**What rule do you want to change?**
prefer-const

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
option for prefer-const to allow mutations to objects.

```js
{
    "prefer-const": ["error", {
        "destructuring": "any",
        "ignoreReadBeforeAssign": false,
        "allowObjectMutation": true
    }]
}
```

**Please provide some example code that this change will affect:**

```js
let newObject = {
  someProp: true
}

newObject['otherProp'] = false

```

**What does the rule currently do for this code?**

Currently generates an error as the object 'newObject' is not reassigned, but rather mutated. If I change the 'let' statement to 'var' the rule doesn't trigger.

**What will the rule do after it's changed?**
No error

**Are you willing to submit a pull request to implement this change?**
Yes, though I'm not familiar with ESLint's codebase, so will take a while for me to figure out.

**Comment by g-plane** [comment]
Thanks for your report.

You still can use `const` at your example. `const` just means it can't be reassigned, but it's mutable for objects.
